### PR TITLE
lnrpc: fully dockerize protobuf generation

### PR DESCRIPTION
**Pull request by ardevd**
Remove the need for golang on the host by utilizing golang docker containers to obtain protobuf and grpc gateway versions.

## Change Description
Improve the dockerized protobuf generation by removing the need for golang on the host system. Makes life easier for those of us who prefers to our host systems clean and use vscode devcontainers or docker containers for their development environment.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [x] Any new logging statements use an appropriate subsystem and logging level.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

### CODE REVIEW

File: lnrpc/gen_protos_docker.sh [hunk 1]
```diff
@@ -5,8 +5,8 @@ set -e
 # Directory of the script file, independent of where it's called from.
 DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
 
-PROTOBUF_VERSION=$(go list -f '{{.Version}}' -m google.golang.org/protobuf)
-GRPC_GATEWAY_VERSION=$(go list -f '{{.Version}}' -m github.com/grpc-ecosystem/grpc-gateway/v2)
+PROTOBUF_VERSION=$(docker run --rm -v $(pwd)/../:/lnd -w /lnd golang go list -f '{{.Version}}' -m google.golang.org/protobuf)
```
Comment (guggero) [inline lnrpc/gen_protos_docker.sh:8]: Should use `$DIR` instead of `$(pwd)` to make sure this works independently of the directory called from and to avoid yet another sub shell to be invoked. Could also extract the `golang` part into a `GO_IMAGE=golang:1.19.7` env variable to make that a bit more clear and pin the version explicitly to the one we use everywhere else.

**Review by guggero — COMMENTED** [review]
Cool, thanks for the update.

File: lnrpc/gen_protos_docker.sh (hunk 1 above)
Comment (guggero) [reply lnrpc/gen_protos_docker.sh]: Yes, unfortunately. But not having a fixed version number has caused issues in the past, so rather have "well defined behavior" instead of things suddenly braking because there's a new Go version out that changes things.

Comment (ardevd) [reply lnrpc/gen_protos_docker.sh]: Totally agree

Comment (ardevd) [reply lnrpc/gen_protos_docker.sh]: Done

**Comment by ardevd** [comment]
@guggero anything left to do here?

File: lnrpc/gen_protos_docker.sh [hunk 2]
```diff
@@ -5,8 +5,13 @@ set -e
 # Directory of the script file, independent of where it's called from.
 DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
 
-PROTOBUF_VERSION=$(go list -f '{{.Version}}' -m google.golang.org/protobuf)
-GRPC_GATEWAY_VERSION=$(go list -f '{{.Version}}' -m github.com/grpc-ecosystem/grpc-gateway/v2)
+# golang docker image version used in this script
```
Comment (guggero) [inline lnrpc/gen_protos_docker.sh:8]: Please use punctuation in comments.

**Review by guggero — COMMENTED** [review]
Can you squash the two commits please?

File: lnrpc/gen_protos_docker.sh (hunk 2 above)
Comment (ardevd) [reply lnrpc/gen_protos_docker.sh]: Done

**Comment by ardevd** [comment]
> Can you squash the two commits please?

Squashed!

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.195.0/containers/go/.devcontainer/base.Dockerfile
+# [Choice] Go version (use -bullseye variants on local arm64/Apple Silicon): 1, 1.16, 1.17, 1-bullseye, 1.16-bullseye, 1.17-bullseye, 1-buster, 1.16-buster, 1.17-buster
+ARG VARIANT=1-bullseye
```
Comment (guggero) [inline .devcontainer/Dockerfile:3]: Was this committed by accident? If not, this should probably be its own PR with a clear rationale why this needs to be in the `lnd` repo.

Comment (ardevd) [reply .devcontainer/Dockerfile]: Yep, that was totally a mistake. I love my devcontainers but sometimes they sneak into commits by mistake. Nice catch. My bad.

**Review by guggero — APPROVED** [review]
LGTM :tada:
